### PR TITLE
Split up GitHub Actions vs Travis CI sections (SOFTWARE-4174)

### PR DESCRIPTION
**Pull request by edquist**

### CODE REVIEW

File: docs/software/container-development-guide.md [hunk 1]
```diff
@@ -8,20 +8,24 @@ This document contains instructions for OSG Technology Team members, including:
...
 
-Creating a New Container Image
-------------------------------
 
-### Prepare the GitHub repository ###
+New Containers Using GitHub Actions
+-----------------------------------
 
-1. Create a Git repository whose name is prefixed with `docker-`, e.g. `docker-frontier-squid`
+### Creating a new container image (GitHub Actions) ###
```
Comment (brianhlin) [inline docs/software/container-development-guide.md:15]:
```suggestion
### Creating a new container image ###
```

Section headers with the same name in a doc are numbered so we can have duplicates

File: docs/software/container-development-guide.md [hunk 2]
```diff
@@ -8,20 +8,24 @@ This document contains instructions for OSG Technology Team members, including:
 - How to manage tags for images in the [OSG DockerHub organization](https://hub.docker.com/r/opensciencegrid/)
 - Tips for container image development
 
-Creating a New Container Image
-------------------------------
 
-### Prepare the GitHub repository ###
+New Containers Using GitHub Actions
+-----------------------------------
```
Comment (brianhlin) [inline docs/software/container-development-guide.md:13]:
```suggestion
New Containers Using GitHub Actions (preferred)
-----------------------------------------------
```

File: docs/software/container-development-guide.md [hunk 3]
```diff
@@ -8,20 +8,24 @@ This document contains instructions for OSG Technology Team members, including:
...
 
-1. Create a Git repository whose name is prefixed with `docker-`, e.g. `docker-frontier-squid`
+### Creating a new container image (GitHub Actions) ###
+
+#### Prepare the GitHub repository (GitHub Actions) ####
```
Comment (brianhlin) [inline docs/software/container-development-guide.md:17]:
```suggestion
#### Prepare the GitHub repository ####
```

File: docs/software/container-development-guide.md [hunk 4]
```diff
@@ -33,7 +37,86 @@ Creating a New Container Image
...
+1. Give write permissions to the "osg-bot" user for this GitHub repo, navigating to:
+   1. "Settings"
+   1. "Manage access"
+   1. "Invite teams or people"
+   1. Search for and select "osg-bot"
+   1. Choose the "Write" role, and click the button to Add osg-bot to the repo.
+   (The osg-bot user needs this permission in order to trigger automatic builds.)
+1. Ask the Software Manager to set up `DOCKER_USERNAME` and `DOCKER_PASSWORD` secrets for this new repo.
+
+#### Set up repository dispatch from docker-software-base (GitHub Actions) ####
```
Comment (brianhlin) [inline docs/software/container-development-guide.md:59]:
```suggestion
#### Set up repository dispatch from docker-software-base ####
```

File: docs/software/container-development-guide.md [hunk 5]
```diff
@@ -33,7 +37,86 @@ Creating a New Container Image
...
+
+1. Edit the
+   [GitHub Actions workflow for docker-software-base](https://github.com/opensciencegrid/docker-software-base/blob/master/.github/workflows/build-container.yml),
+   and add the new GitHub repo name to the `dispatch-repo:` list (under `jobs:` `dispatch:` `strategy:` `matrix:`).
+1. Make a Pull Request for your change.
+
+#### Prepare the DockerHub repository (GitHub Actions) ####
```
Comment (brianhlin) [inline docs/software/container-development-guide.md:66]:
```suggestion
#### Prepare the Docker Hub repository ####
```

Also, move this section above the repo dispatch setup

File: docs/software/container-development-guide.md [hunk 6]
```diff
@@ -33,7 +37,86 @@ Creating a New Container Image
...
+
+1. Create a Docker Hub repo in the OSG organization.
+   The name should generally match the GitHub repo name, without the `docker-` prefix.
+1. Go to the permissions tab and give the `robots` and `technology` teams `Read & Write` access
+
+#### Triggering Container Image Builds (GitHub Actions) ####
```
Comment (brianhlin) [inline docs/software/container-development-guide.md:72]:
```suggestion
#### Triggering Container Image Builds ####
```

File: docs/software/container-development-guide.md (hunk 1 above)
Comment (edquist) [reply docs/software/container-development-guide.md:15]: How do we disambiguate between them in links?

Comment (brianhlin) [reply docs/software/container-development-guide.md]: I forget the exact formatting but this section would look something like `#creating-a-new-container-image1` and the Travis CI subsection could be referred to with `#creating-a-new-container-image2`.

File: docs/software/container-development-guide.md (hunk 5 above)
Comment (edquist) [reply docs/software/container-development-guide.md]: >Also, move this section above the repo dispatch setup

The reason i thought the repo dispatch setup would go first is that that is still part of the github setup (though for the docker-software-base repo), before moving on to docker hub setup.  But anyway i've updated it.

Comment (brianhlin) [reply docs/software/container-development-guide.md]: I'm just worried that the update to `docker-software-base` could get merged, triggering a docker push from the new GitHub repo before the Docker Hub repo is ready.

**Comment by brianhlin** [comment]
CI failures unrelated. Merging.

**Pull request merged**